### PR TITLE
DNS LWRP

### PR DESCRIPTION
**Pull request by rlaveycal**
A new LWRP that manages A and CNAME records in Windows DNS using the DNSCMD command.

### CODE REVIEW

File: providers/dns.rb [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+include Chef::Mixin::ShellOut
+include Windows::Helper
+
+# Support whyrun
+def whyrun_supported?
+  true
+end
+
+action :create do
+	if @current_resource.exists
```
Comment (ksubrama) [inline providers/dns.rb]: Could you change \t to two spaces to be consistent with everything else in the cookbook?

File: providers/dns.rb [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+	parts = @current_resource.host_name.scan(/(\w+)\.(.*)/)
+	@host = parts[0][0]
+	@domain = parts[0][1]
+
+	getAttributes
+end
+
+private
+def getAttributes
+	@command = locate_sysnative_cmd("dnscmd.exe")
```
Comment (ksubrama) [inline providers/dns.rb]: Isn't dnscmd deprecated?  I don't have much experience in the area but there seem to be equivalent powershell commands like Get-DnsServerResourceRecord that provide typed data and lets us avoid parsing command output.

File: README.md [hunk 3]
```diff
@@ -90,6 +91,47 @@ windows_batch 'echo some env vars' do
 end
 ```
 
+### windows_dns
+Configures A and CNAME records in Windows DNS. This requires the DNSCMD to be installed, which is done by adding the DNS role to the server or installing the Remote Server Admin Tools.
+
+#### Actions
+- :create: creates/updates the DNS entry
```
Comment (ksubrama) [inline README.md]: A :install action that simply calls windows_feature to install the dns server might solve the requirement that DNSCMD.exe be installed.

**Comment by ksubrama** [comment]
Thanks a lot for contributing this code!

/cc @smurawski thoughts on the powershell vs dnscmd approach?

File: providers/dns.rb (hunk 2 above)
Comment (rlaveycal) [inline providers/dns.rb]: Get-DnsServerResourceRecord seems to be Windows 2012 only...I need to target 2008R2

**Comment by tas50** [comment]
We should have merged this a while ago. Thanks for the contribution

**Pull request merged**